src/Storages/ExternalStream/Kafka/Kafka.cpp

```diff
@@ -144,6 +144,23 @@ void Kafka::cacheVirtualColumnNamesAndTypes()
     virtual_column_names_and_types.push_back(NameAndTypePair(ProtonConsts::RESERVED_EVENT_SEQUENCE_ID, std::make_shared<DataTypeInt64>()));
 }
 
+klog::KafkaWALAuth Kafka::auth() const
+{
+        return klog::KafkaWALAuth {
+            .security_protocol = settings->security_protocol.value,
+            .username = settings->username.value,
+            .password = settings->password.value,
+            .sasl_mechanism = settings->sasl_mechanism.value,
+            .ssl_ca_cert_file = settings->ssl_ca_cert_file.value,
+        };
+}
+
+klog::KafkaWALSimpleConsumerPtr Kafka::getConsumer(int32_t fetch_wait_max_ms) const
+{
+
+        return klog::KafkaWALPool::instance(nullptr).getOrCreateStreamingExternal(settings->brokers.value, auth(), fetch_wait_max_ms);
```

Comment:
It may not matter, but we have a bunch of memory copy every time we call getConsumer for auth

Reply:
The auth struct is small and this is not in a code hot path, copying looks fine to me. But I changed `auth()` just returns a reference instead anyways.